user_management/api/views.py

```diff
@@ -151,7 +152,8 @@ def initial(self, request, *args, **kwargs):
         try:
             self.user = User.objects.get(pk=uid)
         except User.DoesNotExist:
-            raise Http404()
+            msg = _('Invalid or expired token.')
+            raise PermissionDenied(detail=msg)
```

Comment:
This is what wikipedia is saying about:
 - 400: The server cannot or will not process the request due to something that is perceived to be a client error.
 - 401: Similar to 403 Forbidden, but specifically for use when authentication is required and has failed or has not yet been provided.
 - 403: The request was a valid request, but the server is refusing to respond to it. Unlike a 401 Unauthorized response, authenticating will make no difference.

As not yet authenticated I thought 403 was more appropriate.